src/objc.h

```diff
@@ -0,0 +1,138 @@
+
+/* Compiler implementation of the D programming language
+ * Copyright (c) 2015 by Digital Mars
+ * All Rights Reserved
+ * written by Michel Fortin
+ * http://www.digitalmars.com
+ * Distributed under the Boost Software License, Version 1.0.
+ * http://www.boost.org/LICENSE_1_0.txt
+ * https://github.com/D-Programming-Language/dmd/blob/master/src/objc.h
+ */
+
+#ifndef DMD_OBJC_H
+#define DMD_OBJC_H
+
+#include "root.h"
+#include "mtype.h"
+#include "stringtable.h"
+
+#include "objc_glue.h"
+
+class Identifier;
+class FuncDeclaration;
+class ClassDeclaration;
+class InterfaceDeclaration;
+class ObjcSelector;
+
+struct elem;
```

Comment:
As this is a frontend header, it should avoid using `elem`.  Usually this is done by moving the relevant methods to visitors or free functions, then adding prototypes to the glue `.c` files where necessary.  Remember, frontend headers are common with gdc/ldc which do not use the same glue methods.